### ISSUE TITLE
Anyone has problems since today?

### ISSUE THREAD

**Issue by netmaniac** [jwilk/mbank-cli#61]
Hi!
Today, on few different machines (Debian and Ubuntu) I got errors on any operation:

```
 ./mbank-cli --verbose
* selected command: list
* cookiejar = /home/viciu/.local/share/mbank-cli/XXXXXXX.cookies
* cafile = /usr/share/ca-certificates/mozilla/VeriSign_Class_3_Public_Primary_Certification_Authority_-_G5.crt
* GET https://online.mbank.pl/pl
HTTP error 500 while processing request <GET https://online.mbank.pl/pl>
| Can't connect to online.mbank.pl:443 (certificate verify failed)
| LWP::Protocol::https::Socket: SSL connect attempt failed error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed at /usr/local/share/perl/5.14.2/LWP/Protocol/http.pm line 46.
 at ./mbank-cli line 526
	main::download('HTTP::Request=HASH(0x17859e0)') called at ./mbank-cli line 931
	main::do_login('config', 'HASH(0x18c1fc0)') called at ./mbank-cli line 2485
```
This is new VM with Debian 7.11

```
viciu@debian:~/mbank-cli$ ./mbank-cli --version
mbank-cli 1.6.3
+ LWP::UserAgent 6.26
+ LWP::Protocol::https 6.07
+ IO::Socket::SSL 2.048
+ Net::SSLeay 1.81
  * OpenSSL 1.0.1t  3 May 2016
```

With debug enabled: 
```
cat dbg/pl.html
Can't connect to online.mbank.pl:443 (certificate verify failed)

LWP::Protocol::https::Socket: SSL connect attempt failed error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed at /usr/local/share/perl/5.14.2/LWP/Protocol/http.pm line 46.

cat dbg/pl.html.request 
GET https://online.mbank.pl/pl
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Encoding: gzip, deflate
Accept-Language: en-US,en;q=0.5
User-Agent: Mozilla/5.0 (Windows NT 6.1; rv:45.0) Gecko/20100101 Firefox/45.0

(no content)
```
And when I issue command from shell, I get response:
```
lwp-request -S -d https://online.mbank.pl/pl
GET https://online.mbank.pl/pl
302 Found
GET https://online.mbank.pl/pl/Login
200 OK
```

Switching cacert file to ca.crt within repository does not help.

Any ideas what could be wrong?

**Comment by chilek** [comment]
The same in gentoo current stable:
```sh
# /opt/chilan/mbank-cli --version
mbank-cli 1.6.2
+ LWP::UserAgent 6.15
+ LWP::Protocol::https 6.06
+ IO::Socket::SSL 2.024
+ Net::SSLeay 1.81
  * OpenSSL 1.0.2k  26 Jan 2017
```
**openssl s_client -connect online.mbank.pl:https** works like it should be (I mean it doesn't display any certificate issues).

At the same time the problem doesn't appear in CentOS 7.x.

Termorary workaround is to disabled ssl certificate verification as in patch below:
```patch
--- mbank-cli	2017-05-17 12:36:25.000000000 +0200
+++ mbank-cli	2017-05-17 12:36:39.188266392 +0200
@@ -425,7 +425,7 @@
             # IO::Socket::SSL::set_args_filter_hack('use_defaults') fails to
             # correctly restore SSL_ca_* settings:
             # https://bugs.debian.org/750642
-        SSL_verify_mode => IO::Socket::SSL::SSL_VERIFY_PEER,
+        SSL_verify_mode => IO::Socket::SSL::SSL_VERIFY_NONE,
         SSL_verifycn_scheme => {
             check_cn => 0,
             wildcards_in_alt => 0,
```

**Comment by jwilk** [comment]
CA certificate has changed to *DigiCert High Assurance EV Root CA*.
mbank-cli expects is to be *VeriSign Class 3 Public Primary Certification Authority - G5*, so it now broke.
Sorry for the inconvenience.
I'll fix that shortly.

**Comment by chilek** [comment]
@jwilk isn't it possible to use system wide ca certificates by default?

**Comment by chilek** [comment]
Adding the following line:
```
cafile /etc/ssl/certs/ca-certificates.crt
```
in mbank-cli config file resolves the problem for Gentoo.

**Comment by jwilk** [comment]
I've updated the CA certificate in b2afe5f52007c09bd3421b9d5f1b6b1f3936e4e4.
I'll make a new release soon.
I've also opened #62 to discuss the future of CA pinning.

**Comment by jwilk** [comment]
Fixed in 1.6.3.

**Comment by TheKerimek** [comment]
Thanks for quick solving issue!

Also got error today, now it works fine with Debian 7 wheezy
